syft/execution/plan.py

```diff
@@ -622,6 +636,65 @@ def unbufferize(worker: AbstractWorker, protobuf_plan: PlanPB) -> "Plan":
 
         return plan
 
+    @property
+    def code(self):
+        """Returns string representation of Plan actions"""
+        _self = self
+
+        def stringify(obj, var_name=""):
+            if isinstance(obj, PlaceHolder):
+                if obj.id in input_names:
+                    line = input_names[obj.id]
+                elif obj.id in output_names:
+                    line = output_names[obj.id]
+                elif obj.id in state_names:
+                    line = state_names[obj.id]
+                else:
+                    if obj.id in var_names:
+                        name = var_names[obj.id]
+                    else:
+                        name = len(var_names)
+                        var_names[obj.id] = name
+                    line = f"_{var_name}{name}"
+            elif isinstance(obj, PlaceholderId):
+                line = stringify(_self.role.placeholders[obj.value])
+            elif isinstance(obj, (tuple, list)):
+                line = ", ".join(stringify(o) for o in obj)
+            else:
+                line = str(obj)
+
+            return line
+
+        inputs = self.role.input_placeholders()
+        outputs = self.role.output_placeholders()
+        states = self.role.state.state_placeholders
+        input_names = {ph.id: f"_arg{i+1}" for i, ph in enumerate(inputs)}
+        output_names = {ph.id: f"_out{i+1}" for i, ph in enumerate(outputs)}
+        state_names = {ph.id: f"_state{i+1}" for i, ph in enumerate(states)}
+        var_names = {}
+
+        out = ""
+        out += f"def {self.name}("
+        out += stringify(inputs)
+        out += "):\n"
+        for action in self.role.actions:
```

Comment:
This part seems partially redundant with the new method on `ComputationAction`. Can this delegate to the `__str__` methods on the component classes?

Also, can it be a `__str__` method itself?

Reply:
Some time earlier code representation was removed from `__str__` so I figured it's better to make it separate method.

Reply:
Removed duplication by using communication action's method to create string representation.
Though, it's not very elegant because single communication action doesn't track placeholder's names, so we need to pass variable names.